### ISSUE TITLE
Extension fails to start and uses high CPU

### ISSUE THREAD

**Issue by hotplot** [redhat-developer/vscode-java#472]
When opening a Java project in VS Code, Language Support for Java fails to start and CPU usage goes to 100% on one core

The extension server process continues to run even after quitting Code and needs to be killed manually.

I have verified that the problem is definitely caused by the Language Support for Java extension. With all extensions disabled, the problem does not occur. With this extension alone enabled, CPU usage goes to 100%

I have tried the troubleshooting steps from the wiki:

1) With message tracing enabled, VS Code logs a single message to the language server asking it to initialise (attached below)
2) Developer Tools -> Console is empty (no messages or other output)
3) `Java: Open Java Language Server log file` has no effect - no log file is opened
4) Cleaning the workspace directory has no effect

##### Environment
- Operating System: Ubuntu 16.04
- JDK version: openjdk version "1.8.0_151" (also occurs with openjdk 9)
- Visual Studio Code version: 1.21.1
- Java extension version:

##### Steps To Reproduce
1. Create default gradle project or open single Java file
2. Wait for Language Support for Java to initialise

##### Current Result

Language Support for Java fails to start and instead uses 100% of one CPU core

##### Expected Result

The extension should start successfully

##### Additional Informations

Messages trace: [messages.txt](https://github.com/redhat-developer/vscode-java/files/1823085/messages.txt)

**Comment by fbricon** [comment]
can you provide a threaddump? use `jps` to find the id of a org.eclipse.equinox.launcher*.jar process, then `jstack processid`

**Comment by hotplot** [comment]
No Java processes seem to be running while the CPU is at 100%. I opened a Java project, observed that the CPU increased from idle to 100%, and ran `jps`. The only listed process was jps itself.

I have worked around the issue by doing the following:

    apt purge ^openjdk
    apt autoremove
    rm -rf .vscode/ .config/Code/
    apt install openjdk-8-jdk

I had tried deleting my VS Code data earlier without success, so it is more likely that removing the JDKs fixed the problem for me. I had 8 and 9 installed, and version 8 set as the default. Now I only have JDK 8 installed.

eclipse.jdt.ls says it cannot be compiled using JDK 9. Maybe JDK 9 was being used when the extension started even though JDK 8 was set as my default?

Closing now since I no longer have the issue
